extension/user/login.js

```diff
@@ -0,0 +1,53 @@
+const crypto = require('crypto')
+const {EINVALIDCREDENTIALS, ENOTFOUND, customError} = require('../error')
+/**
+ * @typedef {Object} LoginInputArgs
+ * @property {string} strategy
+ * @property {Object} parameters
+ * @property {string} parameters.login
+ * @property {string} parameters.password
+ */
+
+/**
+ * @param {SDKContext} context
+ * @param {LoginInputArgs} input
+ * @param {function} cb
+ */
+module.exports = function (context, input, cb) {
+  // is logged in
+  if (context.meta.userId) {
+    return cb(null, {
+      userId: context.meta.userId
+    })
+  }
+
+  context.storage.extension.get(input.parameters.login, (errUserId, userId) => {
+    if (errUserId) {
+      return cb(errUserId)
+    }
+
+    if (!userId) {
+      return cb(customError(ENOTFOUND, 'User is not found'))
+    }
+
+    context.storage.extension.get(userId, (err, user) => {
+      if (err) {
+        return cb(err)
```

Comment:
Error should be logged and EINTERNAL should be returned instead.

Reply:
done